### DISCUSSION TITLE
Postgres scanner does not attach partitioned tables

### DISCUSSION THREAD

**Discussion by dpprdan** [duckdb/duckdb#6934] [Q&A] [answered]
After attaching a PostgreSQL DB

```
D CALL postgres_attach('my_connection_string', source_schema='public');
```

I try to query a partitioned table from the Postgres DB and get the following error.

```
D SELECT * FROM partitioned_table LIMIT 10;
Error: Catalog Error: Table with name partitioned_table does not exist!
Did you mean "partitioned_table _0"?
LINE 1: SELECT * FROM partitioned_table LIMIT 10;
```

`PRAGMA show_tables;` indeed does not show the `partioned_table` but (apparently all of) its partitions. 

A query using `postgres_scan()` does succeed however.

```
SELECT * FROM postgres_scan('my_connection_string', 'public', 'partitioned_table') LIMIT 10;

[output from query]
```
(Since I want to join tables I assume that this does not help me much.)

Is this a known problem? Should I file an issue?

**Comment by Mytherin** [comment, marked as answer]
Thanks for the report! I don’t think this is a known issue - please file it in the postgres scanner repository.

All postgres_attach does is create a number of views that call postgres_scan, so you could do that yourself as well as a work-around.

**Reply by dpprdan** [reply]
> All postgres_attach does is create a number of views that call postgres_scan, so you could do that yourself as well as a work-around.

Ah, so 

```
SELECT *
FROM postgres_scan('my_connection_string', 'public', 't1') AS t1
WHERE t1.pk IN (
    SELECT t2.t1_fk
    FROM postgres_scan('my_connection_string', 'public', 't2') AS t2
    );
```
is equivalent to 
```
SELECT *
FROM t1
WHERE t1.pk IN (
    SELECT t2.t1_fk
    FROM t2
    );
```
if `t1` and `t2` can be `postgres_attach()`ed?
